_posts/2019-11-09-using-vault-with-strimzi.md

```diff
@@ -0,0 +1,185 @@
+
+# SSL, Strimzi & Security
+
+## The `incubator` Kafka charts
+
+Like me, you may have tried configuring SSL for Kafka using the `incubator` Helm charts (you can find them [here](https://github.com/helm/charts/tree/master/incubator/kafka)), and realised that in fact this is _far from trivial_. I had initially configured an `initContainer` to get certificates from the Vault PKI backend using the Vault CLI, which did indeed work. However, in addition to the startup scripts in the main broker container and another custom `initContainer` I had configured to set up rack awareness, this caused the the pod startup times to become quite long!
+
+At this point I had configured the brokers and clients with SSL, but what about the Zookeepers? Zookeeper does not support TLS, which is problematic if, for example, you are a financial organisation streaming sensitive data through your Kafka cluster. Even if you're not a fintech company, but you're a sensible individual with security concerns, this might be a deal breaker for you!
+
+## Strimzi & Security
+
+Straight out of the box, [Strimzi provides all of this for you](http://strimzi.io/docs/master/#security-str):
+
+> Strimzi supports encrypted communication between the Kafka and Strimzi components using the TLS protocol. Communication between Kafka brokers (interbroker communication), between Zookeeper nodes (internodal communication), and between these and the Strimzi operators is always encrypted. Communication between Kafka clients and Kafka brokers is encrypted according to how the cluster is configured. For the Kafka and Strimzi components, TLS certificates are also used for authentication.\
+> The Cluster Operator automatically sets up TLS certificates to enable encryption and authentication within your cluster. It also sets up other TLS certificates if you want to enable encryption or TLS authentication between Kafka brokers and clients.
+
+By default, Strimzi generates its own CA and stores the CA certificate and private key as Kubernetes secrets. Using these two secrets, Strimzi generates certificates for all of the components in the cluster, which in turn ensures all communication is encrypted!
+
+## Bringing your own CA
+
+Now, in order to bring your own CA, you need to do this yourself and add the necessary certificate and key as Kubernetes secrets manually, adding the following labels: 
+
+```bash
+  strimzi.io/kind="Kafka"
+  strimzi.io/cluster="${clusterName}"
+```
+
+# Why Vault?
+
+Using Vault to create your own CA arguably gives you more control over your TLS configuration.
+
+Consider a Strimzi Kafka cluster with a Kafka client you have developed. The cluster requires that clients authenticate themselves using SSL, i.e. `ssl.client.auth="required"`.
```

Comment:
It seems that a Kafka cluster deployed by Strimzi, always required that clients have to authenticate via TLS (please use TLS instead of SSL) but it's not true because you have to configure the `tls` listener in the YAML file for the cluster. I think that you can try to be more clear on this sentence.